src/test/java/com/squareup/javapoet/MethodSpecTest.java

```diff
@@ -94,60 +100,82 @@ private TypeElement getElement(Class<?> clazz) {
     @Override public abstract String toString();
   }
 
+  interface ExtendsOthers extends Callable<Integer>, Comparable<Long> {
+    // empty on purpose
+  }
+
   @Test public void overrideEverything() {
     TypeElement classElement = getElement(Everything.class);
     ExecutableElement methodElement = getOnlyElement(methodsIn(classElement.getEnclosedElements()));
-
-    MethodSpec method = MethodSpec.overriding(methodElement).build();
+    ExecutableType type = (ExecutableType) compilation.getTypes().asMemberOf((DeclaredType) classElement.asType(), methodElement);
+    MethodSpec method = MethodSpec.overriding(methodElement, type).build();
     assertThat(method.toString()).isEqualTo(""
         + "@java.lang.Override\n"
+        + "@java.lang.Deprecated\n"
         + "protected <T extends java.lang.Runnable & java.io.Closeable> "
-        + "java.lang.Runnable everything(java.lang.String arg0, java.util.List<? extends T> arg1) "
+        + "java.lang.Runnable everything("
+        + "@com.squareup.javapoet.MethodSpecTest.Nullable java.lang.String arg0, "
+        + "java.util.List<? extends T> arg1) "
         + "throws java.io.IOException, java.lang.SecurityException {\n"
         + "}\n");
-    // TODO see TODOs in MethodSpec.override
-    //assertThat(method.toString()).isEqualTo(""
-    //    + "@java.lang.Override\n"
-    //    + "@java.lang.Deprecated\n"
-    //    + "protected <T extends java.lang.Runnable & java.io.Closeable> "
-    //    + "java.lang.Runnable everything("
-    //    + "@com.squareup.javapoet.MethodSpecTest.Nullable java.lang.String arg0, "
-    //    + "java.util.List<? extends T> arg1) "
-    //    + "throws java.io.IOException, java.lang.SecurityException {\n"
-    //    + "}\n");
   }
 
-  @Ignore // TODO see TODOs in MethodSpec.override
   @Test public void overrideDoesNotCopyOverrideAnnotation() {
-    TypeElement classElement = getElement(Everything.class);
+    TypeElement classElement = getElement(HasAnnotation.class);
     ExecutableElement methodElement = getOnlyElement(methodsIn(classElement.getEnclosedElements()));
-
-    MethodSpec method = MethodSpec.overriding(methodElement).build();
+    ExecutableType type = (ExecutableType) compilation.getTypes().asMemberOf((DeclaredType) classElement.asType(), methodElement);
+    MethodSpec method = MethodSpec.overriding(methodElement, type).build();
     assertThat(method.toString()).isEqualTo(""
         + "@java.lang.Override\n"
-        + "public java.lang.String toString() {"
-        + "}");
+        + "public java.lang.String toString() {\n"
+        + "}\n");
+  }
+
+  @Test public void overrideExtendsOthersWorksWithActualTypeParameters() {
+    TypeElement classElement = getElement(ExtendsOthers.class);
+    List<ExecutableElement> methods = methodsIn(compilation.getElements().getAllMembers(classElement));
+    Iterator<ExecutableElement> iter = methods.iterator();
+    while (iter.hasNext()) {
```

Comment:
Definitely not a fan of while loops in tests. Can you write a test for `call` and a separate test for `compareTo` ?

As is this will pass if the iterator has no elements, or if the methods aren't found